transform/all_collection/spark/transformations/commons.py

```diff
@@ -242,6 +241,22 @@ def harvest_research_community(df: DataFrame, harvested_properties: Dict) -> Non
     harvested_properties[RESEARCH_COMMUNITY] = rc_column
 
 
+def harvest_doi(df: DataFrame, harvested_properties: Dict) -> None:
+    """Harvest DOI from OAG resources"""
+    pids_list = df.select("pid").collect()
+    doi_column = []
+
+    for pids in pids_list:
+        if pids["pid"]:
+            doi_column.append(
+                [pid["value"] for pid in pids["pid"] if pid["scheme"] == DOI]
```

Comment:
Good use of `inline for`. You may also put the `[pid["value"] for pid in pids["pid"] if pid["scheme"] == DOI]` to separate variable to improve readability. It's important to take care of that part since we - software developers - spend 10 times more time reading than writing the code. 

Example solution for better readability:

```python
doi_urls = [pid["value"] for pid in pids["pid"] if pid["scheme"] == DOI]
doi_column.append(doi_urls)

Reply:
Are you sure that the `value`, and `scheme` keys are always available in `pid`? Won't the code throw `KeyError`?

Reply:
You could simplify the code significantly using default values. It's very good practice in the case of transforming arrays.

Source of solution:
https://stackoverflow.com/questions/13710631/is-there-shorthand-for-returning-a-default-value-if-none-in-python

Final example: 

```python
for pids in pids_list:
    pids_values = pids["pid"] or []
    doi_urls = [pid["value"] for pid in pids_values if pid["scheme"] == DOI]
    doi_column.append(doi_urls)
```

When `pids['pid']` is None then `pids_values` will equal `[]`
When `pids_values` is `[]`, then `doi_urls` are equal to `[]`

Reply:
> Are you sure that the `value`, and `scheme` keys are always available in `pid`? Won't the code throw `KeyError`?

Generally, yes. If a `pid` exists the `value` and `schema` keys should always exist too. If that assumption wasn't true, that will mean that OpenAire data is invalid in that case. But I agree that we can be prepared for that.

Reply:
> You could simplify the code significantly using default values. It's very good practice in the case of transforming arrays.
> 
> Source of solution: https://stackoverflow.com/questions/13710631/is-there-shorthand-for-returning-a-default-value-if-none-in-python
> 
> Final example:
> 
> ```python
> for pids in pids_list:
>     pids_values = pids["pid"] or []
>     doi_urls = [pid["value"] for pid in pids_values if pid["scheme"] == DOI]
>     doi_column.append(doi_urls)
> ```
> 
> When `pids['pid']` is None then `pids_values` will equal `[]` When `pids_values` is `[]`, then `doi_urls` are equal to `[]`

Yeah, great idea to use `or` for default values. 
I have applied your suggestion! :clap: